packages/api/src/services/http/http.service.ts

```diff
@@ -138,6 +151,66 @@ export class HttpService {
     return JSON_CONTENT_TYPES.some((contentType) => responseContentType.includes(contentType));
   }
 
+  /**
+   * Performs an HTTP request with the specified method and options.
+   *
+   * @param url     The URL to send the request to.
+   * @param method  The HTTP method to use (GET, POST, PUT, DELETE).
+   * @param options (Optional) An object containing the request options, including:
+   *                 - `params`: Query parameters as key-value pairs.
+   *                 - `headers`: Request headers as key-value pairs.
+   *                 - `body`: The request body.
+   * @returns A Promise that resolves with the full HttpResponse (body of type `T`, status, headers), or is rejected with an error if the request fails.
+   */
+  private requestFull<T>(
+    url: string,
+    method: 'GET' | 'POST' | 'PUT' | 'DELETE' | 'PATCH',
+    options: HttpOptions = {}
+  ): Promise<HttpResponse<T>> {
+    const queryString = this.buildQueryParams(options.params);
+    const fullUrl = `${url}${queryString ? `?${queryString}` : ''}`;
+    const headers = {
+      'Content-Type': 'application/json',
+      Accept: 'application/json, text/plain, */*',
+      ...options.headers,
+    };
+
+    return this.interceptorService
+      .preProcess(new HttpRequest({ url: fullUrl, method, headers, body: options.body }))
+      .then((request) =>
+        fetch(request.url, {
+          method: request.method,
+          headers: request.headers as HeadersInit,
+          body: request.body ? JSON.stringify(request.body) : null,
+        })
+      )
+      .then((response) => this.interceptorService.postProcess(response))
+      .then((response) => {
+        if (!response.ok) {
+          return Promise.reject(response);
+        }
+        const isJson = this.hasValidJson(response);
+        return {
+          json: (isJson ? response.json() : Promise.resolve()),
```

Comment:
Why do you set the `json` prop to a resolved promise?

If the response data is not json it would mess up the logic as the returned object would contain `{... data: Promise}`